adr/2025-07-02-expressions-engine.md

```diff
@@ -0,0 +1,119 @@
+```
+Title: 'Expressions' in funding service forms
+Owner: Samuel Williams
+
+Collaborator(s): Platform Team
+Created on: 2025-07-02
+Status: Final
+Finalised on: 
+```
+
+## Overview
+
+The Funding Service has decided to take on the responsibility for form building through its own service, rather than using some commercial off-the-shelf (COTS) software. This is because form building has been an extreme pain point for us since the service was built, and the available options do not seem to allow us to move with enough velocity and reliability. It's a controversial and significant decision to build our own thing, but we have reasonable confidence that it will result in a better outcome in the long run. When building forms, we need the ability to validate user input and route users to various questions in an easy-but-flexible way. This ADR covers the settled implementation for the Funding Service's "expressions engine", which powers conditions, validation, and personalisation of forms.
+
+## What is the current state?
+
+The expressions engine is a novel implementation for the new funding-service app, so there is limited strictly-current state to compare against. However, there are adjacent parts of MHCLG's and Funding Service's technical estate that achieve similar things to the 'expressions engine' we have built:
+
+### Delta
+
+Delta's form builder and runner is built on [Orbeon Forms](https://www.orbeon.com/). It is extremely powerful and extensible, letting you write essentially any validation logic you want using [XPath expressions](https://doc.orbeon.com/xforms/xpath). MHCLG and the Funding Service does not many people knowledgeable in this DSL, and has dedicated Orbeon Form Engineers to help abstract the complexity of this from the majority of people using Delta for forms. This can be a bottleneck for the service, leading to stress and undue pressure both on the Orbeon Engineers, the service, and the department.
```

Comment:
```suggestion
Delta's form builder and runner is built on [Orbeon Forms](https://www.orbeon.com/). It is extremely powerful and extensible, letting you write essentially any validation logic you want using [XPath expressions](https://doc.orbeon.com/xforms/xpath). MHCLG and the Funding Service does not have many people knowledgeable in this DSL, and has dedicated Orbeon Form Engineers to help abstract the complexity of this from the majority of people using Delta for forms. This can be a bottleneck for the service, leading to stress and undue pressure both on the Orbeon Engineers, the service, and the department.
```